server/server.js

```diff
@@ -1,203 +1,43 @@
-      if (songs.length === 1) {
-        songs.push(...defaultSongs.filter(s => s.uri !== songCurrentlyPlaying.uri));
-      }
-
-      let tracks = [...songs].map(s => s.uri);
-      
-      return spotifyApi.replaceTracksInPlaylist(process.env.SPOTIFY_USER, process.env.SPOTIFY_PLAYLIST, tracks)
-        .then(() => {
-          if (!isJukeboxOn) {
-            setTimeout(() => {
-              spotifyApi.play({
-                context_uri: `spotify:user:${process.env.SPOTIFY_USER}:playlist:${process.env.SPOTIFY_PLAYLIST}`,
-                offset: {
-                  position: 1
-                }
-              })
-              .catch(err => console.log(err))
-            }, 5000);
-          }
-          
-          if (lastPlayed.uri === songCurrentlyPlaying.uri) songs.unshift(lastPlayed);
-          io.emit('update', songs);
-          return songs;
-        })
-        .catch(err => ({ error: 'We couldn\'t add your song for some reason. Try again!', err}));  
-      })
-      .catch(err => ({ error: 'SPACEBOX is turned off. Tell an instructor!', err}))
-}
-
-app.post('/api/request', (req, res) => {
-  if (isDup(req.body.uri)) {
-    res.json({ error: 'Duplicate song!' })
-  }
-  else {
-    spotifyApi.getTrack(req.body.uri.slice(14))
-      .then(track => {
-        const trackData = formatSong(track.body);
-        updatePlaylist(trackData)
-          .then(songs => res.send(songs))
-      })
-      .catch(err => res.json({ error: 'Track doesn\'t exist! Try spotify:track:{SONG_ID}' }));
-  }
-});
-
-//Custom routes
-app.get('/404', (req, res) => res.json({ message: 'Nothing is here. But thanks for checking!' }));
-
-app.use('/api', checkToken);
-app.get('/api/artist/:artist', (req, res) => {
-  spotifyApi.searchArtists(req.params.artist)
-    .then(data => {
-      const items = data.body.artists.items;
-      if (!items.length) res.send('Stop it');
-      res.send(items[0])
-    })
-    .catch(err => console.log(err));
-})
-
-app.get('/api/playlist', async (req, res) => {
-  const songs = await updatePlaylist();
-  res.send(songs);
+  });
+};
+
+// Bootstrap the application, configure models, datasources and middleware.
+// Sub-apps like REST API are mounted via boot scripts.
+boot(app, __dirname, function (err) {
+  if (err) throw err;
+  // start the server if `$ node server.js`
+  if (require.main === module)
+    app.io = require('socket.io')(app.start());
+  app.io.on('connection', function(socket) {
+    socket.on('room', (room) => {
+      socket.join(room);
+      setTimeout(() => {
+        app.io.in(room).emit('update', []);
+      }, 5000);
+    });
+  });
 });
+ 
 
-app.delete('/api/request', (req, res, next) => {
-  spotifyApi.removeTracksFromPlaylist(process.env.SPOTIFY_USER, process.env.SPOTIFY_PLAYLIST, req.body.tracks)
-    .then((response) => {
-      io.emit('update', songs);
-      res.send(response)
+    app.get('/api/spotify/:search', (req, res) => {
```

Comment:
Extract this into its own remote method and use spotify node wrapper

Reply:
This was old and is now GONE